app/scripts/lib/hkdf.js

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+define(['sjcl', 'lib/promise'], function (sjcl, p) {
+  'use strict';
+
+  /**
+   * hkdf - The HMAC-based Key Derivation Function
+   * based on https://github.com/mozilla/node-hkdf
+   *
+   * @class hkdf
+   * @param {bitArray} ikm Initial keying material
+   * @param {bitArray} info Key derivation data
+   * @param {bitArray} salt Salt
+   * @param {integer} length Length of the derived key in bytes
+   * @return promise object- It will resolve with `output` data
+   */
+  function hkdf(ikm, info, salt, length) {
+
+    var mac = new sjcl.misc.hmac(salt, sjcl.hash.sha256);
```

Comment:
This is a copy of our fxa-js-client module: https://github.com/mozilla/fxa-js-client/blob/master/client/lib/hkdf.js